src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts

```diff
@@ -2371,6 +2433,20 @@ baseMipLevelSize: [${baseMipLevelSize.join(', ')}]
 physicalMipCount: ${physicalMipLevelCount}
   `;
 }
+
+function getComponentsToCheck(
+  device: GPUDevice,
+  format: GPUTextureFormat,
+  builtin: TextureBuiltin,
+  textureType: string
+) {
+  const returnsOneComponent = !isBuiltinGather(builtin) && isSingleChannelInput(textureType);
+  const useRGBA =
+    !returnsOneComponent &&
+    (!isDepthOrStencilTextureFormat(format) || device.features.has('texture-component-swizzle'));
+  return useRGBA ? kRGBAComponents : kRComponent;
```

Comment:
nit: Having a hard time reading this logic. Maybe:
```ts
const returnsOneComponent = !isBuiltinGather(builtin) && isSingleChannelInput(textureType);
if (returnsOneComponent) return kRComponent;

const gbaUndefined = 
  isDepthOrStencilTextureFormat(gotFormat) && !device.features.has('texel-component-swizzle');
if (gbaUndefined) return kRComponent;

return kRGBAComponent;
```

Reply:
done